core/core-spacemacs-buffer.el

```diff
@@ -261,15 +262,48 @@ If ALL is non-nil then truly all banners can be selected."
   "Return the full path to banner with index INDEX."
   (concat spacemacs-banner-directory (format "%03d-banner.txt" index)))
 
+(defun spacemacs-buffer//banner-fit-height-size ()
+  "Calculate height of startup banner to fit buffer contents.
+Returns height in units of line height with a minimum of 1."
+  ;; first determine number of lines occupied by startup list
+  (let* ((startup-list-line-height (apply #'+ (mapcar (lambda (l)
+                                                        ;; each section has 2
+                                                        ;; (header and
+                                                        ;; separator) plus
+                                                        ;; number of entries
+                                                        ;; number of lines
+                                                        (+ 2 (cdr l)))
+                                                      dotspacemacs-startup-lists)))
```

Comment:
This works, however it would be much cooler if the scale would be determined by the actual size of the list on the home buffer and not on the potential size. Right now I have 4 projects however my project list is set to store up to 40 resulting in the logo being too small for the actual content on the screen.

Reply:
After checking the code I have noticed this won't be too easy, if you don't want to implement this please put the default scale to "1" for now and I merge it, adding the optimisation later.

Reply:
The first point turned out not to be so much work, so I have implemented that.
As far as I understand your second point, a scaling of 1 is already the default maximum scaling
(see the comment `return factor with maximum of 1` in the `spacemacs-buffer//insert-image-banner` below)

Reply:
Yes however as long as the scale is so aggressiv I would keep the default value in the dotfile to be "1" instead of "auto".

Reply:
Ah okay, but what do you mean with 'aggressive'? I mean, I would call the current behavior 'gentle' (at least how it behaves here, and I hope now also there)...

Reply:
Yeah, what I mean is that the scale of the picture is derived from the potencial size of the lists and not from the actual list content. This is agressive as it may reserve more room for the list of links than what is actually necessary.

Especially I am afraid that the default scale will be too small with a fresh dotfile.

Therefore I would like to have the default value not set to 'auto but to 1 to keep the original scale by default until the scale algorithm is improved not to reserve more space then needed.

Or have this improvement already integrated into this PR.

Reply:
Ah okay, but that is what I meant with 'I have implemented your first point'; I meant that I have adapted the PR so that the scaling is derived from the actual size (In my previous comment I wrote 'startup buffer size' but I meant 'startup buffer list size). So I guess the PR (with the default still set to `auto`) is okay now.

Reply:
Ahhh sorry than this was a big misunderstanding from my side. I am merging your changes.